src/Bugsnag/Payload/StackTraceLine.cs

```diff
@@ -24,19 +24,43 @@ public IEnumerator<StackTraceLine> GetEnumerator()
         yield break;
       }
 
-      var needFileInfo = true;
-      var stackTrace = new System.Diagnostics.StackTrace(_originalException, needFileInfo);
-      var stackFrames = stackTrace.GetFrames();
+      var exceptionStackTrace = true;
+      var stackFrames = new System.Diagnostics.StackTrace(_originalException, true).GetFrames();
+
+#if !NETSTANDARD1_3
+      if (stackFrames == null)
+      {
+        // this usually means that the exception has not been thrown so we need
+        // to try and create a stack trace at the point that the notify call
+        // was made.
+        exceptionStackTrace = false;
+        stackFrames = new System.Diagnostics.StackTrace(true).GetFrames();
+      }
+#endif
 
       if (stackFrames == null)
       {
         yield break;
       }
 
+      bool seenBugsnagFrames = false;
+
       foreach (var frame in stackFrames)
       {
         var stackFrame = StackTraceLine.FromStackFrame(frame);
 
+        if (!exceptionStackTrace)
+        {
+          // if the exception has not come from a stack trace then we need to
+          // skip the frames that originate from inside the notifier code base
+          var currentStackFrameIsNotify = stackFrame.MethodName.StartsWith("Bugsnag.Client.Notify");
```

Comment:
Will all stackframes within the Bugsnag notifier always start with this method name? E.g. would this handle if an exception was thrown internally within the notifier?

Reply:
They won't. The idea here is that if we do not have a stack trace it will be due to notifying of an exception that was not thrown eg. `Bugsnag.Notify(new Exception());` Therefore we are looking for the boundary which is the `Notify` calls as we will have generated a stack trace that includes code from inside the notifier. If an exception was thrown from within the notifier then it would have a stack trace and not fall into this path

Reply:
Got it, that makes sense.